src/web_app/src/pages/operation.tsx

```diff
@@ -4,202 +4,212 @@ import CssBaseline from "@mui/material/CssBaseline";
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+    const gen:number = useLocation().state.nb_epochs;
     const dataset_choice = url.split("/")[2];
-    const method_choice = url.split("/")[4];
     const [
-        method_str,
-        setMethod_str
-    ] = React.useState("");
+        nb_cars,
+        setNb_cars
+    ] = React.useState("1");
     const [
-        data,
-        setData
-    ] = React.useState<d3Types.d3Graph>(
-        {nodes: [], links: []});
+        nb_epochs,
+        setNb_epochs
+    ] = React.useState(1);
     const [
-        data1,
-        setData1
-    ] = React.useState<[Types.Received_data]>([
-        {
-            "time"   : 0,
-            "fitness": 0
-        }
-    ]);
+        colors,
+        setColors
+    ] = React.useState<Array<string>>([]);
     const [
-        data2,
-        setData2
-    ] = React.useState<[Types.Received_data]>([
+        graphdata,
+        setgraphdata
+    ] = React.useState([
         {
-            "time"   : 0,
-            "fitness": 0
+            "id"  : "",
+            "data": [
+                {
+                    "node": 0,
+                    "x"   : 0,
+                    "y"   : 0
+                }
+            ]
         }
     ]);
     const [
-        data3,
-        setData3
-    ] = React.useState<[Types.Received_data]>([
+        nodes,
+        setNodes
+    ] = React.useState<{ [key: string]: [number, number] }>(
         {
-            "time"   : 0,
-            "fitness": 0
+            "0": [
+                0,
+                0
+            ]
         }
-    ]);
+    );
     const [
-        data4,
-        setData4
-    ] = React.useState<[Types.Received_data]>([
+        plotdata,
+        setplotdata
+    ] = React.useState<Array<Types.Point>>([
         {
-            "time"   : 0,
-            "fitness": 0
+            "generation": 0,
+            "fitness"   : 0
         }
     ]);
-    const [
-        start,
-        setStart
-    ] = React.useState<boolean>(false);
     const [
         enableButton,
         setEnableButton
     ] = React.useState(false);
+    const [
+        toggle,
+        setToggle
+    ] = React.useState<boolean>(true);
 
-    const getFirstSol = useCallback(async() => {
-        await http.get(`get_first_sol?${dataset_choice}`)
-            .then(response_f => {
-                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-                setData(response_f.data);
-                if (method_choice === "ga") {
-                    setMethod_str("Genetic Algorithm");
-                } else {
-                    setMethod_str("Deep Reinforcement Learning");
-                }
-            });
-    }, [
-        dataset_choice,
-        method_choice
-    ]);
+    useEffect(() => {
+        if (datasets[parseInt(dataset_choice)]) {
+            setNb_cars(datasets[parseInt(dataset_choice)].split("-")[2][1]);
```

Comment:
if you have say 10 cars, this part of the code will only register 1 car.

Reply:
oh yes. Good point ! I'll fix this.